proposals/nnnn-mt-registration-limitation.md

```diff
@@ -0,0 +1,133 @@
+  * Core will also put the app's HMI level to NONE and notify BC.OnAppUnregistered to HMI after certain timeout if above condition is met while the app is running. (For example, the mandatory transport is disconnected and does not recover within the timeout.)
+
+These features can be configured through smartDeviceLink.ini so that OEMs can enable, disable or customize their behavior.
+
+
+## Detailed design
+
+### Modification of MOBILE\_API.xml
+
+Add `UNAUTHORIZED_TRANSPORT_REGISTRATION` error enum. This value is used by RegisterAppInterface response, so `resultCode` param of RegisterAppInterface response should be updated as well.
+
+```xml
+    <enum name="Result" internal_scope="base">
+
+        :
+
+        <element name="UNAUTHORIZED_TRANSPORT_REGISTRATION">
+            <description>Core doesn't accept registration based on the app's transport type, AppHMIType and OS type. Try registration using another transport.</description>
+        </element>
+```
+
+### Modification of Core
+
+smartDeviceLink.ini includes following new sections. (Note: the default values are provided as an example.)
+
+```ini
+[AcceptedAppHMITypeTable]
+; This section is used to filter apps that can be registered.
+; Apps with AppHMITypes specified in this section are allowed to register.
+; AppHMITypes are specified per transport and OS type. OEM can, for example, reject registration request from specific transport.
+; The AppHMIType(s) is/are specified as comma-separated list. If the list should be empty, specify 'none'.
+; 'empty' is a special word indicating apps that do not specify AppHMIType.
+; '*' indicates all apps, including those that do not have AppHMIType specified.
+WiFi-iOS = none
+WiFi-Android = *
+iAP = *
+Bluetooth-SPP = *
+AOA = *
+
+[DisabledAppHMITypeTable]
+; This section is used to specify apps that can be registered but not allowed to launch.
+; Apps with AppHMITypes specified in this section are allowed to register. However, they will be kept in HMILevel being NONE level and
+; will not be notified to HMI.
+; AppHMITypes are specified per transport and OS type. OEM can, for example, disable specific type(s) of apps which are connected
+; through specific transport.
+; The AppHMIType(s) is/are specified as comma-separated list. If the list should be empty, specify 'none'.
+; 'empty' is a special word indicating apps that do not specify AppHMIType.
+; '*' indicates all apps, including those that do not have AppHMIType specified.
+; If same AppHMIType is specified in both AcceptedAppHMITypeTable and DisabledAppHMITypeTable, the entry in DisabledAppHMITypeTable is
+; ignored and apps with the AppHMIType are launched as usual.
+; also, if an app has multiple AppHMITypes, and they are listed in both AcceptedAppHMITypeTable and DisabledAppHMITypeTable, then the
+; app is launched as usual.
+; If an AppHMIType is not specified to neither of AcceptedAppHMITypeTable and DisabledAppHMITypeTable, then apps with the AppHMIType

```

Comment:
Please update to "...not specified to neither `AcceptedAppHMITypeTable` nor `DisabledAppHMITypeTable`, then..."